### PR TITLE
chore: remove imageWithCachedMetadata

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-271

## What changed (plus any additional context for devs)
while looking into the BSC image color issue i realized that imageWithCachedMetadata functionality was severely broken and also the culprit for the issue. 

when we did the performance work we made an entirely separate coinIcon component that relied on it but the entire time it wasnt working as expected other than tokens in the token list

I've removed the component and replaced it with imgixImage. I also removed all metadata usage and we are now using `usePersistentDominantColorFromImage` which will have the same if not faster perf once @jkadamczyk 's improvements land

This might be a smol perf hit for now but brings up the point that we should be leveraging the image + color work that @welps has done to move this logic entirely off of the client.

## What to test
needs some QA for token icons, especially on the balance screen. I havnt checked android so could use some eyes there

### CODE REVIEW

File: src/components/asset-list/RecyclerAssetList2/FastComponents/FastFallbackCoinIconImage.tsx [hunk 1]
```diff
@@ -1,94 +1,45 @@
...
-            ]}
-          />
-        )}
-
-        {!isLoaded && <View style={sx.fallbackWrapper}>{children()}</View>}
-      </View>
-    );
-  }
-);
+export const FastFallbackCoinIconImage = function FastFallbackCoinIconImage({
```
Comment (estrattonbailey) [inline src/components/asset-list/RecyclerAssetList2/FastComponents/FastFallbackCoinIconImage.tsx:8]: I feel physical relief seeing the deleted code compared to this 🙏

File: src/components/coin-icon/CoinIconFallback.js [hunk 2]
```diff
@@ -14,6 +13,7 @@ import {
   isETH,
   magicMemo,
 } from '@/utils';
+import { IS_ANDROID } from '@/env';
```
Comment (estrattonbailey) [inline src/components/coin-icon/CoinIconFallback.js:16]: Bless

**Review by estrattonbailey — APPROVED** [review]

File: src/components/send/SendAssetFormCollectible.js [hunk 3]
```diff
@@ -63,12 +63,8 @@ export default function SendAssetFormCollectible({
   const [containerWidth, setContainerWidth] = useState();
   const [isGradientVisible, setIsGradientVisible] = useState(false);
 
-  const { dimensions: cachedImageDimensions } = useImageMetadata(
```
Comment (jinchung) [inline src/components/send/SendAssetFormCollectible.js:66]: pls remove `useImageMetadata` import from hooks

File: src/components/settings-menu/DevSection.tsx [hunk 4]
```diff
@@ -318,16 +317,6 @@ const DevSection = () => {
                 loadingStates.clearMmkvStorage && <SettingsLoadingIndicator />
               }
             />
-            <MenuItem
-              leftComponent={<MenuItem.TextIcon icon="📷️" isEmoji />}
-              onPress={clearImageMetadataCache}
-              size={52}
-              titleComponent={
-                <MenuItem.Title
-                  text={lang.t('developer_settings.clear_image_metadata_cache')}
```
Comment (jinchung) [inline src/components/settings-menu/DevSection.tsx:327]: can remove this line from `_english.json` and `_french.json`

File: src/model/migrations.ts [hunk 5]
```diff
@@ -315,7 +313,7 @@ export default async function runMigrations() {
 
   const v8 = async () => {
     logger.log('wiping old metadata');
-    await removeLocal(IMAGE_METADATA);
+    //await removeLocal(IMAGE_METADATA);
```
Comment (jinchung) [inline src/model/migrations.ts:316]: pinged separately

File: src/hooks/useColorForAsset.ts [hunk 6]
```diff
@@ -21,12 +22,15 @@ export default function useColorForAsset(
...
-  ) as any;
+  const { result: imageColor } = usePersistentDominantColorFromImage(
+    maybeSignUri(
+      getUrlForTrustIconFallback(
+        mainnet_address || address!,
+        mainnet_address ? AssetType.token : (type as AssetType)
+      ) || '',
+      { w: 40 }
+    ) || ''
+  );
```
Comment (jkadamczyk) [inline src/hooks/useColorForAsset.ts:33]: @skylarbarrera Would love fro you to try checking CPU usage after and before the change using XCode Instruments, I could record a loom how to use it if you wish

**Review by jkadamczyk — APPROVED** [review]

**Comment by skylarbarrera** [comment]
currently an issue on android that needs resolved before merging this

**Comment by skylarbarrera** [comment]
unfortunately its impossible to grok and follow the issues created form this chance on android so i have opted for a more tightly scoped fix for now. 

eventually we will have to use 1 coinicon component instead of 4 but today is not that day 

replacement is here: https://github.com/rainbow-me/rainbow/pull/4674

**Pull request closed without merging**